File: backend/app/models/schedule.py
from datetime import date, time
from typing import Literal

from pydantic import BaseModel

Subject = Literal["数学", "英语", "物理", "体育", "化学", "语文"]


class Schedule(BaseModel):
    id: str = ""
    institution_id: str = ""
    subject: Subject = "数学"
    teacher_name: str = ""
    start_date: date | None = None
    end_date: date | None = None
    start_time: time | None = None
    end_time: time | None = None
    remarks: str = ""

File: frontend/src/types/schedule.ts
export interface Schedule {
  id: string
  institution_id: string
  subject: string
  teacher_name: string | null
  start_date: string | null
  end_date: string | null
  start_time: string | null
  end_time: string | null
  remarks: string | null
}

export interface ScheduleCreate {
  institution_id: string
  subject: string
  teacher_name?: string
  start_date?: string
  end_date?: string
  start_time?: string
  end_time?: string
  remarks?: string
}

export interface ScheduleUpdate {
  institution_id?: string
  subject?: string
  teacher_name?: string
  start_date?: string
  end_date?: string
  start_time?: string
  end_time?: string
  remarks?: string
}